03_letters/saurabhvyas9/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func letters(s string) map[rune]int {
+	frequencyLetters := make(map[rune]int)
+	for _, str := range s {
+		frequencyLetters[str]++
+	}
+	return frequencyLetters
+}
+
+func sortLetters(m map[rune]int) []string {
+	sortedLetters := make([]string, len(m))
+	tempKey := make([]int, 0, len(m))
+	for key := range m {
+		tempKey = append(tempKey, int(key))
```

Comment:
I appending is more expensive than inserting at known position.
I'd go:
```go
i := 0
keys := make([]int, len(m))
for key := range m {
     keys[i] = int(key)
     i++
}

Reply:
Understood. Changes Done.